src/box/space.c

```diff
@@ -621,7 +641,10 @@ space_create(struct space *space, struct engine *engine,
 	rlist_create(&space->memtx_stories);
 	rlist_create(&space->alter_stmts);
 	space->lua_ref = LUA_NOREF;
-	space->state.is_sync = space->def->opts.is_sync;
+	space->state.is_sync = space->def->opts.is_sync ||
+			       (box_consider_system_spaces_synchronous &&
+				space_is_system(space) &&
+				txn_limbo_has_owner(&txn_limbo));
```

Comment:
In case some async system space is added in the future or if we're upgrading from an old enough schema where some of the async system spaces didn't exist yet this code will make them synchronous.

Reply:
Thanks for the catch! Fixed.

Unfortunately, there was no synchronous replication available in 1.7.6, when the last asynchronous system space was added, so I suggest we just trust this code, since it is pretty simple.